File: catkin_ws/src/pruebas/src/ik_newton_node.cpp
#include "ros/ros.h"
#include "urdf/model.h"




int main (int argc, char ** argv)
{
    std::cout << "Initializing Inverse kinematics node" << std::endl;
    ros::init(argc, argv, "ik_newton_node");
    ros::NodeHandle n;
    ros::Rate loop(10);
    std::string urdf_string;
    std::string joint;

    std::cout << "loading model" << std::endl;
    if(!n.getParam("/robot_description", urdf_string))
    {
        std::cout << "Can't read /robot_description" << std::endl;
        return -1;
    }

    urdf::Model robot_model;
    if (!robot_model.initString(urdf_string)) 
    {
        ROS_ERROR("Can't load robot description from string");
       return -1;
    }
    
    //std::cout << robot_model.getRoot()<< std::endl;
   for (const auto& pair : robot_model.joints_) 
    {
        std::cout << "Joint: " << pair.first << std::endl;
    }

    while(ros::ok())
    {
        ros::spinOnce();
        loop.sleep();
    }
}

File: catkin_ws/src/pruebas/src/ik_newton_node.py
import math
import sys
import rospy
import numpy
import tf
import tf.transformations as tft
import urdf_parser_py.urdf


def get_model_info(joint_names):
    robot_model = urdf_parser_py.urdf.URDF.from_parameter_server()
    joints = []
    transforms = []
    for name in joint_names:
        for joint in robot_model.joints:
            if joint.name == name:
                joints.append(joint)
    for joint in joints:
        T = tft.translation_matrix(joint.origin.xyz)
        R = tft.euler_matrix(joint.origin.rpy[0], joint.origin.rpy[1], joint.origin.rpy[2])
        transforms.append(tft.concatenate_matrices(T,R))
    return joints, transforms

def main():
    print("INITIALIZING INVERSE KINEMATIC NODE - ")
    rospy.init_node("ik_geometric")
    loop = rospy.Rate(40)
    while not rospy.is_shutdown():
        loop.sleep()

if __name__ == '__main__':
    main()